src/main/resources/templates/discussionThreads.html

```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+<html xmlns:th="http://www.thymeleaf.org">
+
+<head th:replace="fragments/navbar :: common-header">
+</head>
+
+<div th:replace="fragments/navbar :: navbar">&nbsp;</div>
+
+<div>
+    <h1>Discussion Threads</h1>
+    <div th:each="thread : ${threads}" class="card">
```

Comment:
Also if you change this line to the below your endpoint will function as advertised.
<div th:each="thread : ${thread}" class="card">